agent/src/softwarecontaineragent.cpp

```diff
@@ -261,8 +303,38 @@ void SoftwareContainerAgent::setGatewayConfigs(const uint32_t &containerID, cons
     }
 }
 
-bool SoftwareContainerAgent::setCapabilities(const uint32_t &containerID, const std::vector<std::string> &capabilities)
+void SoftwareContainerAgent::setGatewayConfigs(const uint32_t &containerID,
+                                               const std::map<std::string, std::string> &configs)
+{
+    GatewayConfiguration parsedConfigs;
+
+    for (auto it=configs.begin(); it!=configs.end(); ++it) {
+        json_error_t error;
+        std::string gwID = it->first;
+        std::string configStr = it->second;
+
+        json_t *jConfigs = json_loads(configStr.c_str(),0,&error);
+        if (!jConfigs) {
```

Comment:
You alternate between using `!ptr` and `ptr == nullptr` please choose one style.